engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -80,24 +80,32 @@ def prepare_scc_announce_request(uri_path, auth, params)
       # to SCC.
       # SCC will make sure to handle the data correctly. This removes the need
       # to adapt here if information send by the client changes.
-      scc_request.body = {
+      scc_req_body = {
         hostname: params['hostname'],
         hwinfo: params['hwinfo'],
-        byos: true
-      }.to_json
+        byos_mode: params['proxy_byos_mode']
+      }
+      if params['proxy_byos_mode'].casecmp('hybrid').zero?
+        scc_req_body[:login] = params['scc_login']
+        scc_req_body[:password] = params['scc_password']
+      end
       scc_request
     end
 
-    def prepare_scc_request(uri_path, product, auth, token, email)
-      scc_request = Net::HTTP::Post.new(uri_path, headers(auth, nil))
+    def prepare_scc_request(uri_path, product, auth, params, mode)
+      params_header = params
+      if mode == 'byos'
+        params_header = nil
+      end
+      scc_request = Net::HTTP::Post.new(uri_path, headers(auth, params_header))
       scc_request.body = {
-        token: token,
+        token: params[:token],
```

Comment:
Should this also have a `|| nil` , like `email` look up below for consistency, or vice versa?

Reply:
if a system is PAYG, and activates an extension that is not free, would only end up on SCC if a regcode is set, or
are we passing any extension activation of a PAYG system to SCC (even without regcode) ?

Maybe there should be a check for the token, earlier than this point for PAYG systems

Reply:
@rjschwei ^

Reply:
Only add on products will be passed to SCC.